java/compaction/compaction-core/src/main/java/sleeper/compaction/job/CompactionJob.java

```diff
@@ -140,6 +140,16 @@ public static final class Builder {
         private Builder() {
         }
 
+        public CompactionJob build() {
+            Objects.requireNonNull(tableId, "tableId must not be null");
+            Objects.requireNonNull(jobId, "jobId must not be null");
+            Objects.requireNonNull(inputFiles, "inputFiles must not be null");
+            Objects.requireNonNull(outputFile, "outputFile must not be null");
+            Objects.requireNonNull(partitionId, "partitionId must not be null");
+            checkDuplicates(inputFiles);
+            return new CompactionJob(this);
+        }
+
```

Comment:
Its a bit unusual to move the build method up here, everywhere else in the project its at the end of the Builder class